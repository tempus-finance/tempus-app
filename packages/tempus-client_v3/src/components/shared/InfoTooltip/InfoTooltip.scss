.tc__info-tooltip-content {
  width: 200px;
  padding: 8px;
}
