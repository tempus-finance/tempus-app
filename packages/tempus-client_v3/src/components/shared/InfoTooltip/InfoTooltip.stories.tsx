import { ComponentStory } from '@storybook/react';
import Icon from '../Icon';

import InfoTooltip from './InfoTooltip';

export default {
  title: 'InfoTooltip',
  component: InfoTooltip,
  argTypes: {},
};

const style = {
  background: 'rgba(0, 0, 0, 0.1)',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  padding: '10px',
};

const Template: ComponentStory<typeof InfoTooltip> = args => (
  <div style={style}>
    <InfoTooltip {...args} />
  </div>
);

export const InfoTooltipClick = Template.bind({});
InfoTooltipClick.args = {
  tooltipContent: 'content',
  openEvent: 'click',
};

export const InfoTooltipMouseover = Template.bind({});
InfoTooltipMouseover.args = {
  tooltipContent: 'content',
  openEvent: 'mouseover',
};

export const InfoTooltipWithErrorIcon = Template.bind({});
InfoTooltipWithErrorIcon.args = {
  tooltipContent: 'content',
  openEvent: 'click',
  iconVariant: 'exclamation-error',
  iconSize: 'large',
};

export const InfoTooltipWithCustomContent = Template.bind({});
InfoTooltipWithCustomContent.args = {
  tooltipContent: <Icon variant="info-bordered" size={96} />,
  openEvent: 'click',
};

export const InfoTooltipWithCustomAnchor = Template.bind({});
InfoTooltipWithCustomAnchor.args = {
  tooltipContent: <Icon variant="info-bordered" size={96} />,
  children: <div>anchor</div>,
};
