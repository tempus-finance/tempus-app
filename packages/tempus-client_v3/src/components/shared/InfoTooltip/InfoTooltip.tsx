import { FC, memo, ReactNode, useMemo } from 'react';
import Typography from '../Typography';
import TooltipWrapper, { OpenEvent } from '../TooltipWrapper/TooltipWrapper';
import { TooltipPlacement } from '../Tooltip/Tooltip';
import Icon, { IconVariant } from '../Icon';

import './InfoTooltip.scss';

export interface InfoTooltipProps {
  tooltipContent: ReactNode | string;
  iconVariant?: IconVariant;
  iconSize?: 'large' | 'medium' | 'small' | 'tiny' | number;
  iconColor?: string;
  openEvent?: OpenEvent;
  placement?: TooltipPlacement;
}

const InfoTooltip: FC<InfoTooltipProps> = props => {
  const {
    tooltipContent,
    iconVariant = 'info-bordered',
    iconSize,
    iconColor,
    placement = 'bottom-center',
    openEvent,
    children,
  } = props;

  const content: ReactNode = useMemo(() => {
    if (typeof tooltipContent === 'string') {
      return (
        <Typography className="tc__info-tooltip-content" variant="body-primary">
          {tooltipContent}
        </Typography>
      );
    }
    return tooltipContent;
  }, [tooltipContent]);
  const anchor = useMemo(
    () => children ?? <Icon variant={iconVariant} size={iconSize} color={iconColor} />,
    [children, iconVariant, iconSize, iconColor],
  );

  return (
    <TooltipWrapper tooltipContent={content} placement={placement} openEvent={openEvent}>
      {anchor}
    </TooltipWrapper>
  );
};

export default memo(InfoTooltip) as FC<InfoTooltipProps>;
