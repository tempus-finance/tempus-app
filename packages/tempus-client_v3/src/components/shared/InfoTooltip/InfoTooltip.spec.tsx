import { fireEvent, render } from '@testing-library/react';
import InfoTooltip, { InfoTooltipProps } from './InfoTooltip';

const defaultProps: InfoTooltipProps = {
  tooltipContent: 'content',
};

const subject = (props: InfoTooltipProps) => render(<InfoTooltip {...props} />);

describe('InfoTooltip', () => {
  it('renders an info tooltip', () => {
    const { container } = subject(defaultProps);

    expect(container).not.toBeNull();

    expect(container).toMatchSnapshot();
  });

  it('renders an info tooltip, click to open to see string content', () => {
    const { container, getByRole, queryByText } = subject(defaultProps);

    const icon = getByRole('button');

    expect(container).not.toBeNull();
    expect(icon).not.toBeNull();

    fireEvent.click(icon);

    const content = queryByText(defaultProps.tooltipContent as string);

    expect(content).not.toBeNull();

    expect(content).toMatchSnapshot();
  });

  it('renders an info tooltip, click to open to see content', () => {
    const props = { ...defaultProps, tooltipContent: <div>content</div> };
    const { container, getByRole } = subject(props);

    const icon = getByRole('button');

    expect(container).not.toBeNull();
    expect(icon).not.toBeNull();

    fireEvent.click(icon);

    expect(container).toMatchSnapshot();
  });
});
