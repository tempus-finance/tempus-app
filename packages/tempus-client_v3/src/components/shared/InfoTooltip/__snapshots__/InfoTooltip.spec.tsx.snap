// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoTooltip renders an info tooltip 1`] = `
<div>
  <button
    class="tc__btn tc__tooltip-wrapper-anchor"
    data-selected="false"
    type="button"
  >
    <svg
      class="tc__icon tc__icon-info-bordered"
      fill="none"
      height="24"
      viewBox="0 0 16 16"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M8 15C11.866 15 15 11.866 15 8C15 4.13401 11.866 1 8 1C4.13401 1 1 4.13401 1 8C1 11.866 4.13401 15 8 15ZM8 16C12.4183 16 16 12.4183 16 8C16 3.58172 12.4183 0 8 0C3.58172 0 0 3.58172 0 8C0 12.4183 3.58172 16 8 16Z"
        fill="#222222"
        fill-rule="evenodd"
      />
      <path
        d="M8.9998 10.9969L8.9998 8.01403C8.9998 7.45985 8.55213 7.0106 7.9999 7.0106C7.44759 7.0106 6.99989 7.45997 7 8.01423L7.0006 10.9971C7.00071 11.5401 7.43147 11.9845 7.97243 11.9996C8.53515 12.0153 8.9998 11.5618 8.9998 10.9969Z"
        fill="#222222"
      />
      <path
        d="M7.0003 5.00303C7.00002 5.55732 7.44771 6.00706 8.00005 6.00706C8.55219 6.00706 8.9998 5.55763 8.9998 5.00353C8.9998 4.44944 8.55219 4 8.00005 4C7.4481 4 7.00057 4.44913 7.0003 5.00303Z"
        fill="#222222"
      />
    </svg>
  </button>
</div>
`;

exports[`InfoTooltip renders an info tooltip with customized anchor 1`] = `
<div>
  <button
    class="tc__btn tc__tooltip-wrapper-anchor"
    data-selected="false"
    type="button"
  >
    <div>
      anchor
    </div>
  </button>
</div>
`;

exports[`InfoTooltip renders an info tooltip, click to open to see content 1`] = `
<div>
  <div
    class="tc__tooltip-wrapper"
    style="--anchorPosX: 0px; --anchorPosY: 0px; --anchorVisibility: block;"
  >
    <div
      class="tc__tooltip-wrapper-backdrop"
    />
    <button
      class="tc__btn tc__tooltip-wrapper-anchor"
      data-selected="false"
      type="button"
    >
      <svg
        class="tc__icon tc__icon-info-bordered"
        fill="none"
        height="24"
        viewBox="0 0 16 16"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M8 15C11.866 15 15 11.866 15 8C15 4.13401 11.866 1 8 1C4.13401 1 1 4.13401 1 8C1 11.866 4.13401 15 8 15ZM8 16C12.4183 16 16 12.4183 16 8C16 3.58172 12.4183 0 8 0C3.58172 0 0 3.58172 0 8C0 12.4183 3.58172 16 8 16Z"
          fill="#222222"
          fill-rule="evenodd"
        />
        <path
          d="M8.9998 10.9969L8.9998 8.01403C8.9998 7.45985 8.55213 7.0106 7.9999 7.0106C7.44759 7.0106 6.99989 7.45997 7 8.01423L7.0006 10.9971C7.00071 11.5401 7.43147 11.9845 7.97243 11.9996C8.53515 12.0153 8.9998 11.5618 8.9998 10.9969Z"
          fill="#222222"
        />
        <path
          d="M7.0003 5.00303C7.00002 5.55732 7.44771 6.00706 8.00005 6.00706C8.55219 6.00706 8.9998 5.55763 8.9998 5.00353C8.9998 4.44944 8.55219 4 8.00005 4C7.4481 4 7.00057 4.44913 7.0003 5.00303Z"
          fill="#222222"
        />
      </svg>
    </button>
    <div
      class="tc__tooltip tc__tooltip__bottom-center"
      style="--anchorWidth: 0px;"
    >
      <div>
        content
      </div>
    </div>
  </div>
</div>
`;

exports[`InfoTooltip renders an info tooltip, click to open to see string content 1`] = `
<div
  class="tc__info-tooltip-content"
  style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
>
  content
</div>
`;
